src/main/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1798,31 +1800,37 @@ public long getUptime() {
     }
 
     @Override
-    public CompositeData getToppartitions(String sampler, List<String> keyspaceFilters, List<String> tableFilters, int duration, int capacity, int count) throws OpenDataException {
+    public Map<String, CompositeData> getToppartitions(List<String> keyspaceFilters, List<String> tableFilters, int duration, int capacity, int count) throws OpenDataException {
```

Comment:
Is it possible to keep the old signature too? So that old nodetool works with new jmx. Keep in mind that it's possible to run nodetool remotely, so the packages may not be in sync.

If jmx matches by full signature, not just name, then it should just work.

Reply:
Also: maybe pass a list of samplers, in case Scylla grows the ability to select samplers. This will remove the need to change the signature once more in the future.